iceoryx_posh/test/integrationtests/test_roudi_findservice.cpp

```diff
@@ -71,8 +71,8 @@ TEST_F(RoudiFindService_test, OfferServiceWithAnyServiceIdStringDescriptionFails
 
 TEST_F(RoudiFindService_test, OfferServiceWithAnyServiceIdDescriptionFails)
 {
-    auto isServiceOffered =
-        senderRuntime->offerService(iox::capro::ServiceDescription(iox::capro::AnyService, iox::capro::AnyInstance));
+    auto isServiceOffered = senderRuntime->offerService(
+        iox::capro::ServiceDescription(iox::capro::AnyServiceString, iox::capro::AnyInstanceString));
```

Comment:
In my opinion this shows that this design is not intuitive at all:  Reading this it appears we are not offering a concrete service, but one with wildcards. But this makes no sense...

We need a separate pattern matching class or algorithm for those AnyStrings, ideally we can get rid of them in the class logic.

I know that this is not part of the PR, but we really should consider this in a redesign.

Reply:
This test case is removed in #854. Also see [this comment](https://github.com/eclipse-iceoryx/iceoryx/pull/852#discussion_r652905346)